### PR TITLE
Add multi tenant support

### PR DESCRIPTION
**Pull request by renkelvin**

### CODE REVIEW

File: FirebaseAuth/Sources/Auth/FIRAuth.m [hunk 1]
```diff
@@ -1975,6 +1976,16 @@ - (BOOL)updateCurrentUser:(nullable FIRUser *)user
     [self possiblyPostAuthStateChangeNotification];
     return YES;
   }
+  if (user) {
+    if ((user.tenantID != nil && ![self.tenantID isEqualToString:user.tenantID]) ||
+        (self.tenantID != nil && ![user.tenantID isEqualToString:self.tenantID])) {
```
Comment (rosalyntan) [inline FirebaseAuth/Sources/Auth/FIRAuth.m:1981]: Can this be simplified to `if ((user.tenantID || self.tenantID) && ![self.tenantID isEqualToString:user.tenantID])`?

File: FirebaseAuth/Sources/Utilities/FIRAuthErrorUtils.m [hunk 2]
```diff
@@ -901,6 +919,10 @@
       return @"ERROR_REJECTED_CREDENTIAL";
     case FIRAuthErrorCodeMissingOrInvalidNonce:
       return @"ERROR_MISSING_OR_INVALID_NONCE";
+    case FIRAuthErrorCodeTenantIDMismatch:
+      return @"ERROR_TENANT_ID_MISMATCH";
+    case FIRAuthErrorCodeUnsupportedTenantOperation:
+      return @"UNSUPPORTED_TENANT_OPERATION";
```
Comment (rosalyntan) [inline FirebaseAuth/Sources/Utilities/FIRAuthErrorUtils.m:925]: Should this be `@"ERROR_UNSUPPORTED_TENANT_OPERATION"`?

File: FirebaseAuth/Sources/Auth/FIRAuth.m (hunk 1 above)
Comment (renkelvin) [reply FirebaseAuth/Sources/Auth/FIRAuth.m:1981]: Yeah, looks better.

File: FirebaseAuth/Sources/Utilities/FIRAuthErrorUtils.m (hunk 2 above)
Comment (renkelvin) [reply FirebaseAuth/Sources/Utilities/FIRAuthErrorUtils.m:925]: yeah, good catch.

**Pull request merged**